packages/app/cypress/e2e/cypress-in-cypress-component.cy.ts

```diff
@@ -118,20 +119,30 @@ describe('Cypress In Cypress CT', { viewportWidth: 1500, defaultCommandTimeout:
   })
 
   it('browser picker in runner calls mutation with current spec path', () => {
-    cy.__incorrectlyVisitAppWithIntercept()
+    cy.visitApp()
     cy.contains('TestComponent.spec').click()
     cy.get('[data-model-state="passed"]').should('contain', 'renders the test component')
 
-    cy.intercept('mutation-VerticalBrowserListItems_SetBrowser').as('setBrowser')
+    cy.withCtx((ctx, o) => {
+      o.sinon.stub(ctx.actions.app, 'setActiveBrowserById')
+      o.sinon.stub(ctx.actions.project, 'launchProject').resolves()
+    })
 
     cy.get('[data-cy="select-browser"]')
     .click()
 
     cy.contains('Firefox')
     .click()
 
-    cy.wait('@setBrowser').then(({ request }) => {
-      expect(request.body.variables.specPath).to.contain('/cypress-in-cypress/src/TestComponent.spec.jsx')
+    cy.withCtx((ctx, o) => {
```

Comment:
One general comment - if you're seeing any flake where there's race conditions between the action being fired in response to an event, and the `withCtx` body evaluating on the server, there's a `withRetryableCtx` which can be used in situations where the `withCtx` body is idempotent and just needs to pass with the assertions once.